src/Http/Ownership/SuggestOwnershipsRequestHandler.php

```diff
@@ -0,0 +1,148 @@
+use CultuurNet\UDB3\Search\Sorting;
+use CultuurNet\UDB3\User\CurrentUser;
+use CultuurNet\UDB3\User\UserIdentityResolver;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+final class SuggestOwnershipsRequestHandler implements RequestHandlerInterface
+{
+    private ResultsGeneratorInterface $resultsGenerator;
+    private OfferJsonDocumentReadRepository $offerRepository;
+    private CurrentUser $currentUser;
+    private UserIdentityResolver $userIdentityResolver;
+    private DocumentRepository $organizerRepository;
+    private OwnershipSearchRepository $ownershipSearchRepository;
+    private OrganizerIDParser $organizerIDParser;
+    private UserPermissionsReadRepositoryInterface $userPermissionsRepository;
+
+    public function __construct(
+        SearchServiceInterface $searchService,
+        OfferJsonDocumentReadRepository $offerRepository,
+        CurrentUser $currentUser,
+        UserIdentityResolver $userIdentityResolver,
+        DocumentRepository $organizerRepository,
+        OwnershipSearchRepository $ownershipSearchRepository,
+        OrganizerIDParser $organizerIDParser,
+        UserPermissionsReadRepositoryInterface $userPermissionsRepository
+    ) {
+        $this->resultsGenerator = new ResultsGenerator(
+            $searchService,
+            new Sorting('modified', 'desc')
+        );
+        $this->offerRepository = $offerRepository;
+        $this->currentUser = $currentUser;
+        $this->userIdentityResolver = $userIdentityResolver;
+        $this->organizerRepository = $organizerRepository;
+        $this->ownershipSearchRepository = $ownershipSearchRepository;
+        $this->organizerIDParser = $organizerIDParser;
+        $this->userPermissionsRepository = $userPermissionsRepository;
+    }
+
+    public function handle(ServerRequestInterface $request): ResponseInterface
+    {
+        $this->guardAccess();
+
+        $queryParams = new QueryParameters($request);
+
+        $queryParams->guardRequiredEnum('itemType', [ItemType::organizer()->toString()]);
+
+        $id = $this->currentUser->getId();
+        $user = $this->userIdentityResolver->getUserById($id);
+
+        $query = (new SuggestOwnershipsSapiQuery($user))->toString();
+
+        $activeOwnerships = $this->ownershipSearchRepository
+            ->search(new SearchQuery([
+                new SearchParameter('ownerId', $user->getUserId()),
+            ]))
+            ->filter(self::activeOwnerships());
+
+        /**
+         * A map to deduplicate returned organizers
+         * @var array<string, array> $idToOrganizerMap
+         */
+        $idToOrganizerMap = [];
+
+        /**
+         * @var ItemIdentifier $result
+         */
+        foreach ($this->resultsGenerator->search($query) as $result) {
+            $offerType = OfferType::fromCaseInsensitiveValue($result->getItemType()->toString());
```

Comment:
I wonder if we even need to know the type. It is possible to write the query so that only events are returned and then you only need to access the JSON projections of events.

Reply:
Do we want to only return organizers linked to events? The ticket mentioned fetching from offers endpoint.
The `OfferJsonDocumentReadRepository` requires the `OfferType` as the first argument

Reply:
You are right, we need to take into account events and places.